frontend/src/modules/EditZing/Components/EmailModalContent.tsx

```diff
@@ -0,0 +1,151 @@
+import { useState } from 'react'
+import { EmailModalContentProps } from '../Types/ComponentProps'
+import {
+  ModalContainer as ContentContainer,
+  RecipientsContainer as InfoContainer,
+} from '../Styles/Email.style'
+import { EmailTemplateButtons } from './EmailTemplateButtons'
+import { EmailPreview } from './EmailPreview'
+import { Box, Button, Typography } from '@mui/material'
+import { TemplateName } from 'EditZing/utils/emailTemplates'
+import SendIcon from '@mui/icons-material/Send'
+
+export const EmailModalContent = ({
+  selectedGroups,
+  courseNames,
+  isEmailing,
+  setIsEmailing,
+}: EmailModalContentProps) => {
+  const [selectedTemplate, setSelectedTemplate] = useState<TemplateName>(
+    '' as TemplateName
+  )
+  const [step, setStep] = useState<number>(0)
+  const titles = ['Select email template', 'Send emails']
+
+  const RecipientsComponent = () => {
+    let groupNames = ''
+    selectedGroups.forEach((group, index) => {
```

Comment:
tried to use .reduce here but unfortunately the return type is not of type Group, so :/

Reply:
```js
const groupNames = selectedGroups
  .map(group => `Group ${group.groupNumber}`)
  .join(', ')
```